### ISSUE TITLE
Generated Angular 2 project does not build

### ISSUE THREAD

**Issue by nureyev** [jeddict/jeddict#222]
I tried a simple generated Angular 2 project multiple times but it does not build. 

To stop some errors, I switched the project configuration to 'webpack'. I also updated Maven and NodeJS which allowed the build process to get to the end of the output log that you see below. Previously it failed on more errors. Now it stops at this point:

Output log:

cd C:\Users\Nureyev\Documents\Java\Project_June2017\Project; "JAVA_HOME=C:\\Program Files\\Java\\jdk1.8.0_131" "M2_HOME=C:\\Users\\Nureyev\\Documents\\Java\\apache-maven-3.5.0" cmd /c "\"\"C:\\Users\\Nureyev\\Documents\\Java\\apache-maven-3.5.0\\bin\\mvn.cmd\" -Dmaven.ext.class.path=\"C:\\Program Files\\NetBeans 8.2\\java\\maven-nblib\\netbeans-eventspy.jar\" -Dfile.encoding=UTF-8 -Pwebpack,webpack install\""
Scanning for projects...

------------------------------------------------------------------------
Building Project 1.0-SNAPSHOT
------------------------------------------------------------------------

--- maven-dependency-plugin:2.6:copy (default) @ Project ---

--- frontend-maven-plugin:1.3:install-node-and-yarn (install node and yarn) @ Project ---
Node v6.9.2 is already installed.
Yarn 0.18.1 is already installed.

--- frontend-maven-plugin:1.3:yarn (yarn install) @ Project ---
Running 'yarn install' in C:\Users\Nureyev\Documents\Java\Project_June2017\Project
yarn install v0.18.1
info No lockfile found.
[1/4] Resolving packages...
warning angular2-infinite-scroll@0.3.4: Please use the new --> ngx-infinite-scroll <--- supports in Angular +4, AOT compilation and optimized code bundle
warning browser-sync > localtunnel > request > node-uuid@1.4.8: Use uuid module instead
[2/4] Fetching packages...
warning fsevents@1.1.2: The platform "win32" is incompatible with this module.
info "fsevents@1.1.2" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning Incorrect peer dependency "@angular/common@^2.0.0".
warning Incorrect peer dependency "@angular/core@^2.0.0".
warning Incorrect peer dependency "@angular/core@^2.0.0".
warning Incorrect peer dependency "@angular/http@^2.0.0".
[4/4] Building fresh packages...
success Saved lockfile.
$ yarn run webpack:build

**Comment by nureyev** [comment]
This may help also. It's the output log of "npm install":

Output log:

"C:\Program Files\nodejs\npm.cmd" "install"
npm WARN deprecated angular2-infinite-scroll@0.3.4: Please use the new --> ngx-infinite-scroll <--- supports in Angular +4, AOT compilation and optimized code bundle
npm WARN deprecated node-uuid@1.4.8: Use uuid module instead
npm ERR! Cannot read property 'write' of null
npm WARN angular2-cookie@1.2.6 requires a peer of @angular/common@^2.0.0 but none was installed.
npm WARN angular2-cookie@1.2.6 requires a peer of @angular/core@^2.0.0 but none was installed.
npm WARN ng2-translate@4.2.0 requires a peer of @angular/core@^2.0.0 but none was installed.
npm WARN ng2-translate@4.2.0 requires a peer of @angular/http@^2.0.0 but none was installed.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: istanbul-instrumenter-loader@2.0.0 (node_modules\istanbul-instrumenter-loader):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, utime 'C:\Users\Nureyev\Documents\Java\Project_June2017\Project\node_modules\.staging\istanbul-instrumenter-loader-b60a1137\index.js'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: weinre@2.0.0-pre-I0Z7U9OV (node_modules\weinre):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, open 'C:\Users\Nureyev\Documents\Java\Project_June2017\Project\node_modules\.staging\weinre-02621ae7\web\client\AuditRules.js'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @angular/tsc-wrapped@4.0.0 (node_modules\@angular\tsc-wrapped):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, utime 'C:\Users\Nureyev\Documents\Java\Project_June2017\Project\node_modules\.staging\@angular\tsc-wrapped-1506ab92\src\collector.js'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: less@2.7.2 (node_modules\less):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, open 'C:\Users\Nureyev\Documents\Java\Project_June2017\Project\node_modules\.staging\less-1b3dcae6\test\less\errors\functions-8-element.less'

npm ERR! code EINTEGRITY
npm ERR! sha1-KcVzMUgFe7Th913zW3qcty5qWd0= integrity checksum failed when using sha1: wanted sha1-KcVzMUgFe7Th913zW3qcty5qWd0= but got sha1-f4gDJJKON+OoIbFQ895x+Ff1S8g=. (10511 bytes)

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Nureyev\AppData\Roaming\npm-cache\_logs\2017-06-28T22_41_51_679Z-debug.log
Done.

**Comment by jGauravGupta** [comment]
Just select the maven "webpack" profile and build the project , you don't need to fire  the command manually : https://jeddict.github.io/page.html?mg=ANGULAR4

**Comment by nureyev** [comment]
Sorry. I tried this on more than a dozen times but nothing seems to work. I reset the npm cache and installed angular-cli again but the project can not build. I may try again later. Thanks for your help.

**Comment by jGauravGupta** [comment]
I am not able to reproduce this issue. Please find the attached log of successful build.
[NG2-BUILD.txt](https://github.com/jeddict/jeddict/files/1117580/NG2-BUILD.txt)

Can google you following error, I am not much sure about it   : 
```
npm ERR! code EINTEGRITY
npm ERR! sha1-KcVzMUgFe7Th913zW3qcty5qWd0= integrity checksum failed when using sha1: wanted sha1-KcVzMUgFe7Th913zW3qcty5qWd0= but got sha1-f4gDJJKON+OoIbFQ895x+Ff1S8g=. (10511 bytes)

npm ERR! A complete log of this run can be found in:
npm ERR! C:\Users\Nureyev\AppData\Roaming\npm-cache_logs\2017-06-28T22_41_51_679Z-debug.log
```

**Comment by nureyev** [comment]
If I did this correctly, the troublesome module is 'esrecurse'

This is from the ...\node_modules\esrecurse\package-lock.json

      "version": "2.8.29",
      "resolved": "https://registry.npmjs.org/uglify-js/-/uglify-js-2.8.29.tgz",
      "integrity": "sha1-KcVzMUgFe7Th913zW3qcty5qWd0=",
      "dev": true,
      "optional": true,

This is from ...\node_modules\uglify-js\package.json

  "_id": "uglify-js@2.8.29",
  "_inBundle": false,
  "_integrity": "sha1-KcVzMUgFe7Th913zW3qcty5qWd0=",
  "_location": "/uglify-js",
  "_phantomChildren": {

To solve, the module must be reinstalled. I will attempt this a bit later.

Thank you for your help.

**Comment by jGauravGupta** [comment]
Issue closed as current release (v4.8.2) supports Angular 5
